lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -135,7 +136,7 @@ export class TokenCache implements ITokenCache {
         }
 
         const scopes = new ScopeSet(request.scopes).printScopes();
-        const expiresOn = response.expires_in;
+        const expiresOn = options.expiresOn || response.expires_in;
```

Comment:
Should the right half of this operation be updated to calculate `expiresOn` from `expires_in`?

Reply:
Potentially -- we'd have to add the request timestamp somewhere in scope to make the calculation. This could go into `ServerAuthorizationTokenResponse` (a root for the `ExternalTokenResponse` type) or even `LoadTokenOptions`.

It's worth noting though that the interface creating the "real" tokens is calling `createAccessTokenEntity` and setting expiresOn directly as the callsite has request timing in context already.

I was a bit hesitant to make a change to the real behavior to fix a bug that only occurs in the loading of external tokens; open to suggestions though. Another consideration is potentially making `expiresOn` required or adding `requestTimestamp` into `LoadTokenOptions` to make it clearer to consumers of `loadExternalTokens` that those data are required to get correct timing behavior.

Reply:
Since this is merely for testing purposes, I wonder if we can just use now (or now minus a buffer, say 5 minutes) as the request timestamp. Presumably your test suite would be requesting a token, then immediately loading it, using it once and then moving on to the next test. We don't need to be super accurate for this scenario and for those that prefer a more accurate timestamp they can still provide it explicitly with your change here. Thoughts? @jasonnutter also curious your thoughts here

Reply:
I would say default to now and allow an explicit request timestamp to be optionally passed in.

Reply:
So something like `const expiresOn = options.expiresOn || now + response.expires_in;`?

Reply:
@tnorling @jasonnutter Ok finally got around to pushing - https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/4711/commits/9788d91a41c3adcde125045e0b5a8050f12f7635